workflow/targets/preprocess.smk

```diff
@@ -130,25 +63,39 @@ PREPROCESS_OUTPUT_MAPPINGS = {
 }
 PREPROCESS_OUTPUTS_MAPPED = map_outputs(PREPROCESS_OUTPUTS, PREPROCESS_OUTPUT_MAPPINGS)
 
-# Generate sbs preprocessing targets
-PREPROCESS_OUTPUTS_SBS = {
-    rule_name: templates
-    for rule_name, templates in PREPROCESS_OUTPUTS.items()
-    if "sbs" in rule_name
+# Generate targets using the appropriate wildcard combinations
+# Metadata extraction uses metadata-specific wildcards
+PREPROCESS_TARGETS_SBS_METADATA = outputs_to_targets(
+    {"extract_metadata_sbs": PREPROCESS_OUTPUTS["extract_metadata_sbs"]},
+    sbs_metadata_wildcard_combos,
+    PREPROCESS_OUTPUT_MAPPINGS
+)
+
+PREPROCESS_TARGETS_PHENOTYPE_METADATA = outputs_to_targets(
+    {"extract_metadata_phenotype": PREPROCESS_OUTPUTS["extract_metadata_phenotype"]},
+    phenotype_metadata_wildcard_combos,
+    PREPROCESS_OUTPUT_MAPPINGS
+)
```

Comment:
Have you tested on large amounts of data to see if there are performance issues? I remember running into some when I dynamically created targets with so much code like this before.